src/Extension/MemberExtension.php

```diff
@@ -39,4 +46,68 @@ public function getDefaultRegisteredMethod()
     {
         return $this->owner->RegisteredMFAMethods()->byId($this->owner->DefaultRegisteredMethodID);
     }
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->removeByName(['DefaultRegisteredMethodID', 'HasSkippedMFARegistration', 'RegisteredMFAMethods']);
+
+        if (!$this->currentUserCanViewMFAConfig()) {
+            return $fields;
+        }
+
+        $fields->addFieldToTab(
+            'Root.Main',
+            $methodListField = RegisteredMFAMethodListField::create(
+                'MFASettings',
+                _t(__CLASS__ . '.MFA_SETTINGS_FIELD_LABEL', 'Multi-factor Authentication Settings (MFA)'),
+                $this->owner
+            )
+        );
+
+        // Administrators can't modify registered method
+        if ($this->currentUserCanEditMFAConfig()) {
+            $methodListField->setReadonly(true);
```

Comment:
I'm getting readOnly: true on the field when I view my own profile (`/admin/myprofile`) - should it be false in that case?

Reply:
I guess the condition needs to be flipped. It's false for other members, but should be true (readonly) for other members and false (editable) for yourself